restorer.go

```diff
@@ -39,6 +40,9 @@ func (r *Restorer) Restore(cache Cache) error {
 		}
 	}
 
+	old := archive.SetUmask(0)
```

Comment:
Do the tests fail here if these are omitted?

Reply:
Forgive me, I'm still not sure why these lines are necessary, if `Extract()` is behaving correctly  - and avoiding race conditions. 😅

Reply:
If we do #722 only then we will need to bring back these lines from #203 (and potentially make the extract slower): https://github.com/buildpacks/lifecycle/pull/203/files#diff-64f1e499a23830d0c26ee246125d18431a7e36ad3249c5df70eb2982effc68afR150-R153

Reply:
@jabrown85 yes they fail... rather than unset the umask during the extract, I made it the responsibility of the calling function, otherwise there is no way that this function could be threadsafe (unless we explicitly set the permissions like in https://github.com/buildpacks/lifecycle/pull/203/files#diff-64f1e499a23830d0c26ee246125d18431a7e36ad3249c5df70eb2982effc68afR150-R153 )